### PR TITLE
graph: permissioning

### PR DESCRIPTION
**Pull request by liam-fitzgerald**
- Refactors group-store to make tag `[app=term =resource tag=term]`
- Uses this refactored tag to implement a permissioning system for graph-store, based on mark conversion to the relevant type from the validator mark
-  Reworks the interface for adjusting settings on a channel 
-  Reworks the group settings dialog
Capture d’écran 2021-01-28 à 3 13 47 PM
Capture d’écran 2021-01-28 à 3 14 32 PM
Capture d’écran 2021-01-28 à 3 14 50 PM

-  Enables the member-metadata functionality in the new metadata-store
-  Refactors ShipSearch and modal functionality

### CODE REVIEW

**Comment by urcades** [comment]
so exciting, wow

**Comment by matildepark** [comment]
Conflict here @liam-fitzgerald

**Comment by liam-fitzgerald** [comment]
@tacryt-socryp review pls

File: pkg/arvo/app/graph-push-hook.hoon [hunk 1]
```diff
@@ -124,3 +134,136 @@
...
+  ^-  permission-level:store
+  ?:  is-admin
+    admin.permissions
+  ?:  =(~ writers)
+    writer.permissions
+  ?:  (~(has in writers) src.bowl)
+    writer.permissions
+  reader.permissions
+::
+++  is-allowed
```
Comment (tacryt-socryp) [inline pkg/arvo/app/graph-push-hook.hoon:187]: ??? This doesn't look right.

**Review by tacryt-socryp — COMMENTED** [review]
One part of the logic looks really weird, let me know what's going on there. Otherwise looks good.

File: pkg/arvo/app/graph-push-hook.hoon (hunk 1 above)
Comment (liam-fitzgerald) [reply pkg/arvo/app/graph-push-hook.hoon:187]: lmfao it's dead code, removing

**Review by tacryt-socryp — APPROVED** [review]

**Pull request merged**